src/main/webapp/common.js

```diff
@@ -104,15 +109,19 @@ function addMenu() {
   delivery_request.href = 'deliveryRequest.html';
   menuElement.append(delivery_request);
   const see_journeys = document.createElement('a');
-  let displayJourneys;
-  fetch('/user-data').then(response => response.json()).then((userInfo) => {
-    console.log(userInfo);
-    displayJourneys = userInfo.isCourier;
+  firebase.auth().onAuthStateChanged(function(user) {
```

Comment:
This javascript code should have tests I think.

Otherwise how do you know that this code does what you expect it to do (and will continue to do so in the future?).